src/com/zetcode/Board.java

```diff
@@ -85,26 +94,20 @@ public void paintComponent(Graphics g) {
     }
 
     private void drawObjects(Graphics2D g2d) {
-
-        g2d.drawImage(ball.getImage(), ball.getX(), ball.getY(),
-                ball.getImageWidth(), ball.getImageHeight(), this);
-        g2d.drawImage(paddle.getImage(), paddle.getX(), paddle.getY(),
-                paddle.getImageWidth(), paddle.getImageHeight(), this);
+        
+        ball.draw(g2d);
```

Comment:
Although it may sound like good OOP to place the draw logic directly in the ball class, it is not good if you ever want to port the game.  Game logic should be separate from UI logic.